packages/core/content-type-builder/admin/src/components/ContentTypeBuilderNav/useContentTypeBuilderMenu.js

```diff
@@ -1,26 +1,16 @@
-/**
- *
- * LeftMenu
- *
- */
-
-import React, { useMemo } from 'react';
-import PropTypes from 'prop-types';
-import { sortBy, camelCase, upperFirst } from 'lodash';
-import { useHistory } from 'react-router-dom';
-import { useIntl } from 'react-intl';
-import { LeftMenuList, useTracking, useNotification } from '@strapi/helper-plugin';
 import { Text } from '@buffetjs/core';
+import { useNotification, useTracking } from '@strapi/helper-plugin';
+import { camelCase, isEmpty, sortBy, toLower, upperFirst } from 'lodash';
+import matchSorter from 'match-sorter';
+import React, { useMemo, useState } from 'react';
```

Comment:
Can you import it first?

Reply:
Also can you add tests to this hook it seems that there is a lot of complex logic and I would love to see some tests.

Reply:
To be fair I just c/c a 2 old years code to this hook, so I have no idea what it is doing exactly either ^^'

Reply:
ok fine by me

What about adding some tests to the ContentTypeBuilderNav? We can start by generating a snapshot?